core/primitives/src/epoch_sync.rs

```diff
@@ -25,6 +27,31 @@ pub struct EpochSyncProof {
     pub current_epoch: EpochSyncProofCurrentEpochData,
 }
 
+const MAX_UNCOMPRESSED_EPOCH_SYNC_PROOF_SIZE: u64 = 500 * 1024 * 1024;
+const EPOCH_SYNC_COMPRESSION_LEVEL: i32 = 3;
+
+#[derive(
+    Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, derive_more::From, derive_more::AsRef,
+)]
+pub struct CompressedEpochSyncProof(Box<[u8]>);
+impl
+    CompressedData<
+        EpochSyncProof,
+        MAX_UNCOMPRESSED_EPOCH_SYNC_PROOF_SIZE,
+        EPOCH_SYNC_COMPRESSION_LEVEL,
+    > for CompressedEpochSyncProof
+{
+}
+
+impl Debug for CompressedEpochSyncProof {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("CompressedEpochSyncProof")
+            .field("len", &self.0.len())
+            .field("proof", &self.decode())
```

Comment:
sanity check: Is there any risk for a zip bomb exploding here when logging some error?

Reply:
Well the decode() does potentially return 500MB of data. So yes there is that kind of risk, but this is for printing as part of testloop test; nobody should be printing a whole epoch sync proof in the regular code... even the normal case would look like a zip bomb (43MB or so).